packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/link/read.rs

```diff
@@ -20,138 +16,6 @@ use crate::{
-            "#,
-            &[&source_entity_id],
-        )
-        .await
-        .into_report()
-        .change_context(QueryError)
-        .attach_printable(source_entity_id)?
-        .into_iter()
-        .map(|row| {
-            (
-                VersionedUri::new(
-                    BaseUri::new(row.get(0)).expect("invalid BaseUri"),
-                    row.get::<_, i64>(1) as u32,
-                ),
-                OutgoingLinkTarget::Single(row.get(2)),
-            )
-        }))
-}
-
-#[expect(dead_code, reason = "Multi links are not supported yet")]
-async fn multi_by_source_entity_id(
-    client: &(impl GenericClient + Sync),
-    source_entity_id: EntityId,
-) -> Result<impl Iterator<Item = (VersionedUri, OutgoingLinkTarget)> + Send, QueryError> {
-    Ok(client
-        .query(
-            r#"
-            WITH aggregated as (
-                SELECT link_type_version_id, ARRAY_AGG(target_entity_id ORDER BY multi_order ASC) as links
-                FROM links
-                WHERE active AND multi and source_entity_id = $1
-                GROUP BY link_type_version_id
-            )
-            SELECT base_uri, "version", links FROM aggregated
-            INNER JOIN ids ON ids.version_id = aggregated.link_type_version_id
-            "#,
-            &[&source_entity_id],
-        )
-        .await
-        .into_report()
-        .change_context(QueryError)
-        .attach_printable(source_entity_id)?
-        .into_iter()
-        .map(|row| (
-            VersionedUri::new(BaseUri::new(row.get(0)).expect("invalid BaseUri"), row.get::<_, i64>(1) as u32),
-            OutgoingLinkTarget::Multiple(row.get(2))
-        )))
-}
-
-#[expect(dead_code, reason = "Multi links are not supported yet")]
-async fn by_source_entity_id(
-    client: &(impl GenericClient + Sync),
-    source_entity_id: EntityId,
-) -> Result<Vec<OutgoingLinks>, QueryError> {
-    let single_links = single_by_source_entity_id(client, source_entity_id).await?;
-    let multi_links = multi_by_source_entity_id(client, source_entity_id).await?;
-    Ok(vec![OutgoingLinks::new(
-        single_links.chain(multi_links).collect(),
-    )])
-}
-
-#[expect(dead_code, reason = "Multi links are not supported yet")]
-async fn by_link_type_by_source_entity_id(
-    client: &(impl GenericClient + Sync),
-    link_type_base_uri: &BaseUri,
-    link_type_version: u32,
-    source_entity_id: EntityId,
-) -> Result<Vec<OutgoingLinks>, QueryError> {
-    let link =
-        client
-        .query_one(
-            r#"
-            -- Gather all single-links
-            WITH single_links AS (
-                SELECT link_type_version_id, target_entity_id
-                FROM links
-                INNER JOIN ids ON ids.version_id = links.link_type_version_id
-                WHERE active AND NOT multi AND source_entity_id = $1 AND base_uri = $2 AND "version" = $3
-            ),
-            -- Gather all multi-links
-            multi_links AS (
```

Comment:
I'm keeping some of the queries here to some degree. But I'll figure out where it fits in on my PR